checkmarx/util.py

```diff
@@ -411,3 +415,40 @@ def component_logger(artifact_name: str):
 def create_checkmarx_client(checkmarx_cfg_name: str):
     cfg_fac = ci.util.ctx().cfg_factory()
     return checkmarx.client.CheckmarxClient(cfg_fac.checkmarx(checkmarx_cfg_name))
+
+
+def insert_results(
+    scans: model.FinishedScans,
+    scan_artifacts,
+    compliancedb_cfg_name: str,
+):
+    if not compliancedb_cfg_name:
+        logger.warning('no compliance db cfg name found, skipping insertion')
+        return
+
+    client = ccc.delivery.default_client_if_available()
+
+    for scan_artifact in scan_artifacts:
+        for scan in scans.failed_scans + scans.scans_above_threshold + scans.scans_below_threshold:
```

Comment:
The threshold is not really considered here since in summary, those three lists contain all scan results for checkmarx.
For email / log printing this separation is required. Thus either checkmarx needs some refactoring to extract the separation logic from `checkmarx.util.scan_sources`, call `insert_results()` and split into those three lists afterwards, or, as suggested, use the concatenation.